shapeshift/web
# feat: fox farming old contracts

## Description

this PR adds support for expired v1-3 fox contracts.

## Notice




## Pull Request Type


## Issue (if applicable)

closes #2468

## Risk

none

## Testing

1. enable the `FoxFarming` flag2. 
2. if you have a balance in an expired contract, skip to step 4. if not deposit into fox farming v4 ( you need to make sure `ConsiderFoxFarmingV4ExpiredForTesting` feature flag is off to be able to deposit into v4.) 
3. enable  the `ConsiderFoxFarmingV4ExpiredForTesting` feature flag
4. open up an expired opportunity
5. you should just have the permission to withdraw and claim
6. for re-testing you can use steps 2 & 3 instructions.7. 
 

## Screenshots (if applicable)
<img width="1220" alt="Screen Shot 2022-08-19 at 21 50 04" src="https://user-images.githubusercontent.com/20498757/185681006-7aa303c3-117b-4b2b-8f65-3ad14aecf896.png">
<img width="413" alt="Screen Shot 2022-08-19 at 21 55 50" src="https://user-images.githubusercontent.com/20498757/185681009-83edf9b6-de49-474f-9db6-9dfea66dc892.png">
<img width="509" alt="Screen Shot 2022-08-19 at 21 48 04" src="https://user-images.githubusercontent.com/20498757/185680970-e20f6dee-0268-48e0-acf3-431f79d2a9e2.png">
<img width="523" alt="Screen Shot 2022-08-19 at 21 48 12" src="https://user-images.githubusercontent.com/20498757/185681001-3cc21246-ad24-4443-bec4-8657aac1a4b4.png">
